src/core/stream_recv.c

```diff
@@ -380,14 +450,23 @@ QuicStreamProcessStreamFrame(
         goto Error;
     }
 
-    if (EndOffset > Stream->RecvMaxLength) {
+    if (EndOffset > Stream->RecvMaxLength && !Stream->Flags.RemoteCloseResetReliable) {
         //
-        // Frame goes past the FIN.
+        // Frame goes past the FIN, and the stream is not reset reliably.
         //
         Status = QUIC_STATUS_INVALID_PARAMETER;
         goto Error;
     }
 
+      if (Stream->RecvBuffer.BaseOffset >= Stream->RecvMaxLength && Stream->Flags.RemoteCloseResetReliable) {
```

Comment:
I think we might be able to merge this and the previous if with sub-if scopes to reduce the number of checks that have to be made.

Reply:
Yeah, that makes sense, for performance :)

Reply:
Updated!